include/pybind11/pytypes.h

```diff
@@ -950,6 +950,8 @@ inline namespace literals {
 inline str operator"" _s(const char *s, size_t size) { return {s, size}; }
 }
 
+template <> inline bool isinstance<str>(handle obj) { return PyUnicode_Check(obj.ptr()); }
```

Comment:
This will change the behavior of pybind11, no?

Reply:
Yes. I wrote the long Request for Review comment for exactly this reason. I'm pointing this out in at the beginning ("desirable/undesirable") and at the end ("good or bad?"), hoping to provide a start for collectively reasoning out if, how, and when we want to change the behaviors.

There are actually "3 and a half" behaviors that are (unfortunately) convoluted:

1. isinstance<str> (reminder: `str` is always `actual unicode` in pybind11), keep or match native?
2. implicit decode when passing `bytes` to `pybind11::str`, keep or disable?
3. what do we want to do about `bytes` to `list` conversions in A. Python 2, B. Python 3?

What would be your preferred answers for these 3 cases **independently**?
I guess we can always engineer a way to achieve any combination of preferred answers, even making the changes independently in multiple pybind11 releases.